### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by brian-learningpool**
- Scrambled HTML on hovering project title
- Thumbnail API now returns generic image rather than error

### CODE REVIEW

File: frontend/src/core/app/helpers.js [hunk 1]
```diff
@@ -190,6 +186,14 @@ define(function(require){
 
         },
 
+        ifImageIsCourseAsset: function(url, block) {
+          if (url.length !== 0 && url.indexOf('course/assets') == 0) {
+            return block.fn(this);;
```
Comment (dennis-learningpool) [inline frontend/src/core/app/helpers.js]: couple of double semi-colons here and on line 193

File: frontend/src/core/app/helpers.js [hunk 2]
```diff
@@ -190,6 +186,14 @@ define(function(require){
 
         },
 
+        ifImageIsCourseAsset: function(url, block) {
```
Comment (dennis-learningpool) [inline frontend/src/core/app/helpers.js]: the name of this is a bit weird. can you change it to something like "fixAssetURL" or something a bit more imperative

File: lib/assetmanager.js [hunk 3]
```diff
@@ -28,7 +28,8 @@ var MODNAME = 'assetmanager',
...
...xveYxweY1xe45yfZB0fZF2gJN6hJZ6hZZ8hph9h5iBiZuDjJ6FjqCGj6CGkKGHkKKIkqSKk6WLlKWQmaqTnK2Una2Vna6Xn6+YobGZorKcpbSep7efqLihqrqkrLylrr2qssGwuca1vMq6ws+8xdLAyNXBydbFzdrHz9zI0N3K0t/L0t/M1ODN1ODQ2OPS2uXS2+bV3OjW3ejY3+ra4u3c5O/e5fDf5vHg6PPh6fTj6/bk6/bl7Pfm7ffm7vjn7vno7/np8PoQVD/gAAACnklEQVR4Ae3WbVMSURjG8bNUbmaY1EZqkgpYJhlFUWRkQhKgaQ+JmQ/hw/39P0MzNbfT5l4ehGv3Re3/9XXm9+rM3Ebky6sHXqr/vNnF5S2xZaR21QzexMqBxXljOI3Wuuc6Vwyryc55DrGRVjSOcZvYiQgybMji3MhdPAxhpyT20NvswmUAcZ2SbHoAIjvyLQ0grgMgugMgugMgugMgugMgugMgugMgugMgvgMgvgMgvgMgvgMgvgMgugOgFtsB0EiH7QBoskt1MFSjOhgaPaA6GFoJx1HI08FEqI5sXtLFVqiOLOhimeLcKYGyulikOPZmB3ISPTveQM7tnp3UQE4hIqftRONIOSLnR9lhO6BWIZ0YzFmvZMZcdyxTWZf+y9mc9pQ5baodlnNYNL6Kh6E4u9Pmr6Z3Q3COsuZM2SOqg39Fme5sOEGOs8F25kxgc2Rn24C2uU4dOXWuU0BOgevMIGeG66SRk+Y648gZ5zp55OS5ThE5Ra7TQE6D6+wNBTNDexLQx2fHfTn4uCgHMtfNIwBZnZ3hIGZ4J5gxALI7shrkrCIGQXZHKmeZCmYQZHek6vgVp4oZDNkdaST/ZJINwFgguyP7S8lTZWkfMwiyO9rJWvXp/PyT12snghkMYQcGGAAxHMAoxHMAoxDPAYxCPAcwCvEcwCjEciCjEMdBDIB6dJpNzGDowk7TdZt2BkJ2RxljfNAnxADI7iijEGYwhJ0ARiHAgJ7bHT+jEGJAOcEOYBRShusoo5AyRMfHKFRXhuAARqM6gOE7gOE7gOE7t0w0Tip2Yid2Yuffc/jFTuz8H07mbRh5fie8InZuRuTkI3JeROMkPoh5GIHzUsR8vx8687grYkTe3XVDRK7dey/yyxE5/vo5rDryu5+fSKzliGdNYwAAAABJRU5ErkJggg==';
```
Comment (dennis-learningpool) [inline lib/assetmanager.js]: Can't we just ship a default image on file, and read it from disk? Seems like the default thumbnail is something a designer should be able to change, and we wouldn't lose the chunk of ram that storing this string will take up.

File: frontend/src/core/app/helpers.js (hunk 2 above)
Comment (brian-learningpool) [inline frontend/src/core/app/helpers.js]: I'm not thrilled about the name myself but this function is a handlebars helper and the name is consistent with the other similar helpers in the same file, e.g. `ifValueEquals()`, `ifUserIsMe()`.  Handlebars doesn't support the ability to call a function with an `if` statement in front.

Comment (dennis-learningpool) [inline frontend/src/core/app/helpers.js]: Possibly those functions need named better too - the main problem I have is that the the function name looks like it tests something, but actually modifies the arguments.

File: lib/assetmanager.js (hunk 3 above)
Comment (brian-learningpool) [inline lib/assetmanager.js]: I've raised this as an enhancement (#850) for when we finally get a designer involved in the project.

**Comment by dennis-learningpool** [comment]
+1

File: frontend/src/core/app/helpers.js (hunk 2 above)
Comment (taylortom) [inline frontend/src/core/app/helpers.js]: This doesn't modify anything though does it?

Given: 
```
{{#customIf}}
    <div class="ifDiv"></div>
{{else}}
    <div class="elseDiv"></div>
{{/customIf}}
```
Retuning `block.fn(this)` will insert the `ifDiv`, whereas returning `block.inverse(this)` will insert the `elseDiv` (as I understand it).

**Comment by darylhedley** [comment]
+1

File: frontend/src/core/app/helpers.js (hunk 2 above)
Comment (dennis-learningpool) [inline frontend/src/core/app/helpers.js]: Hi @taylortom - you're correct. I misunderstood the handlebars helper.

**Pull request merged**